fe/fe-core/src/main/java/com/starrocks/server/AbstractTableFactory.java

```diff
@@ -0,0 +1,27 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.server;
+
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.Table;
+import com.starrocks.common.DdlException;
+import com.starrocks.sql.ast.CreateTableStmt;
+
+import javax.validation.constraints.NotNull;
+
+public interface AbstractTableFactory {
+    @NotNull
+    Table createTable(LocalMetastore metastore, Database database, CreateTableStmt stmt) throws DdlException;
```

Comment:
How about putting LocalMetastore into OlapTableFactory

Reply:
Do you mean refactor the signature to `Table createTable(LocalMetastore metastore, Database database, CreateTableStmt stmt)`?

Reply:
yes, but manually creating the external table is currently only for compatibility.  we will recommend using the catalog to query external table. So I think if the this part is mainly for inner tables in future, passing it to localMetastore is fine.